components/script/iframe_collection.rs

```diff
@@ -0,0 +1,163 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use std::cell::Cell;
+use std::default::Default;
+
+use base::id::BrowsingContextId;
+use euclid::{Scale, Size2D};
+use fnv::FnvHashMap;
+use script_layout_interface::IFrameSizes;
+use script_traits::{IFrameSizeMsg, WindowSizeData, WindowSizeType};
+use style_traits::CSSPixel;
+use webrender_api::units::DevicePixel;
+
+use crate::dom::bindings::inheritance::Castable;
+use crate::dom::bindings::root::DomRoot;
+use crate::dom::htmliframeelement::HTMLIFrameElement;
+use crate::dom::node::{Node, ShadowIncluding};
+use crate::dom::types::Document;
+use crate::script_thread::with_script_thread;
+
+#[derive(JSTraceable, MallocSizeOf)]
+pub(crate) struct IFrame {
+    pub element: DomRoot<HTMLIFrameElement>,
```

Comment:
This should be `Dom`(it will require adding `#[allow(crown::unrooted_must_root)]` to the struct, and implies some other changes like mapping it to and from a dom root).

Reply:
>#[allow(crown::unrooted_must_root)]

You mean `#[crown::unrooted_must_root_lint::must_root]`.

Reply:
Makes sense. I've made this change.